app/layout/CmsLayout.tsx

```diff
@@ -0,0 +1,191 @@
+import type { IronSessionUser } from "$types/auth";
+import type { User } from "@prisma/client";
+import type { PropsWithChildren, ReactElement } from "react";
+import type { IconType } from "react-icons/lib";
+import { gql, useMutation, useQuery } from "@apollo/client";
+import { MoonIcon, SunIcon } from "@chakra-ui/icons";
+import {
+	Box,
+	Button,
+	Center,
+	Flex,
+	Heading,
+	Icon,
+	Skeleton,
+	Stack,
+	Text,
+	useColorMode,
+	useColorModeValue,
+} from "@chakra-ui/react";
+import { useRouter } from "next/router";
+import { FiHome, FiFileText, FiUsers, FiImage } from "react-icons/fi";
+import { Link } from "$app/links";
+
+interface NavElementProps {
+	href: string;
+	name: string;
+	selected: boolean;
+	icon: IconType;
+}
+
+function NavElement({ href, name, selected, icon }: NavElementProps) {
+	const defaultColor = useColorModeValue("gray.700", "gray.400");
+
+	const color = selected ? "white" : defaultColor;
+	const background = selected ? "brand.500" : "transparent";
+
+	return (
+		<Link
+			href={href}
+			py={2}
+			borderRadius="lg"
+			bg={background}
+			_hover={{
+				bg: useColorModeValue("gray.400", "gray.800"),
+			}}>
+			<Flex alignItems="center" ml={3}>
+				<Center mr={2}>
+					<Icon color={color} boxSize={5} as={icon} />
+				</Center>
+				<Text
+					fontSize={18}
+					textDecoration="none"
+					fontWeight={600}
+					color={color}>
+					{name}
+				</Text>
+			</Flex>
+		</Link>
+	);
+}
```

Comment:
This looks large enough to put it in a separate file (in general, it should be one component per file anyway)